hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -421,3 +460,31 @@ func generateSSHSecret(data []byte, namespace string) (*corev1.Secret, error) {
 	secret.Type = corev1.SecretTypeOpaque
 	return secret, nil
 }
+
+func namespacedName(obj metav1.Object) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+}
+
+func parseNamespacedName(name string) types.NamespacedName {
+	parts := strings.SplitN(name, string(types.Separator), 2)
+	if len(parts) > 1 {
+		return types.NamespacedName{Namespace: parts[0], Name: parts[1]}
+	}
+	return types.NamespacedName{Name: parts[0]}
+}
+
+func enqueueParentHostedCluster(obj ctrlclient.Object) []reconcile.Request {
+	var hostedClusterName string
+	if obj.GetAnnotations() != nil {
+		hostedClusterName = obj.GetAnnotations()[hostedClusterAnnotation]
+	}
+	if hostedClusterName == "" {
+		return []reconcile.Request{}
+	}
+	return []reconcile.Request{
+		{NamespacedName: parseNamespacedName(hostedClusterName)},
```

Comment:
why do we need `parseNamespacedName` here? wouldn't `namespacedName` be enough?

Reply:
can also probably use `ObjectKeyFromObject` controller runtime helper instead of `namespacedName`

Reply:
We need parseNamespacedName to go from a string (annotation value) to a NamespacedName